tests/recipes/cpc2/test_evaluate.py

```diff
@@ -0,0 +1,119 @@
+"""Tests for the CPC2 evaluation functions."""
+
+import math
+
+import numpy as np
+import pytest
+
+from recipes.cpc2.baseline.evaluate import (
+    compute_scores,
+    kt_score,
+    ncc_score,
+    rmse_score,
+    std_err,
+)
+
+
+@pytest.mark.parametrize(
+    "x, y, expected", [([1, 2, 3], [1, 2, 3], 0), ([0], [1], 1), ([1, 1, 1], [1], 0)]
+)
+def test_rmse_score_ok(x, y, expected):
+    """Test the function rmse_score valid inputs"""
+    assert rmse_score(np.array(x), np.array(y)) == pytest.approx(expected)
+
+
+@pytest.mark.parametrize(
+    "x, y, expected", [([1, 2, 3], [2, 3], ValueError), ([], [], ValueError)]
+)
+def test_rmse_score_error(x, y, expected):
```

Comment:
As with `test_parse_cec2_signal_name()` tests I think these can be combined.

Untested...

```python
@pytest.mark.parametrize(
    "x, y, expected",
    [
        ([1, 2, 3], [1, 2, 3], 0),
        ([0], [1], 1),
        ([1, 1, 1], [1], 0),
        ([1, 2, 3], [2, 3], ValueError),
        ([], [], ValueError)
    ]
)
def test_rmse_score_ok(x, y, expected):
    """Test the function rmse_score valid inputs"""
    with x, y:
        assert rmse_score(np.array(x), np.array(y)) == pytest.approx(expected)
```

Although it may not in this instance as the `expected` is being passed as an argument to `pytest.approx()`. :thinking: 

```

Reply:
Yes, think it is clearer and simpler in the end to keep them separate. Also it means the types stay consistent, ie. (list, list, float) vs (list, list, Exception)